NatLibFi/ekirjasto-android-core
# Implement support for digital magazines

Adds a new module simplified-ekirjasto-magazines, which adds support for digital magazines. The digital magazine service is shown in the app using a WebView, and another fullscreen WebView is used as a magazine reader. If the user is not logged in, a message about logging in is displayed.

There are still some things I'd like to improve and refactor, but this at least *works*, so I'll do those in a later PR.

To test the functionality, try at least these:
- when not signed in, the magazines tab should show text prompting the user to sign in
- when you sign in, the magazines tab should show the magazine browser
- clicking on a magazine shows a magazine volume/issue and clicking on one of those opens a fullscreen reader (fullscreen apart from the system top and bottom bars)
- changing device orientation from portrait to landscape and vice versa should work in the reader - orientation should be locked back to portrait when exiting the reader
- logging out should again show the text prompting the user to sign in

**NOTE:** Some fields were added to the authentication document, and old app data on an Android device will cause issues. All developers and testers should clear old app data (if the app were in production, this would be handled more gracefully inside the app).

Issue: [SIMPLYE-225](https://jira.lingsoft.fi/browse/SIMPLYE-225)
